Models/Core/ApsimFile/Converter.cs

```diff
@@ -1385,6 +1385,21 @@ private static void UpgradeToVersion70(JObject root, string fileName)
             }
         }
 
+        /// <summary>
+        /// Alters all existing linint functions to have a child variable reference IFunction called XValue instead of a
+        /// string property called XProperty that IFunction then had to locate
+        /// </summary>
+        /// <param name="root">The root JSON token.</param>
+        /// <param name="fileName">The name of the apsimx file.</param>
+        private static void UpgradeToVersion71(JObject root, string fileName)
+        {
+            foreach (JObject linint in JsonUtilities.ChildrenRecursively(root, "LinearInterpolationFunction"))
+            {
+                JsonUtilities.AddVariableReference(linint,"XValue", linint["XProperty"].ToString());
```

Comment:
Could probably remove this AddVariableReference function altogether and just do something like this:

```suggestion
                VariableReference variable = new VariableReference();
                variable.Name = "XValue";
                variable.VariableName = linint["XProperty"].ToString();
                JsonUtilities.AddModel(linint, variable);
```

Reply:
I tried doing something like that for a start but could not get it working.  Don't really understand the converter well enough.  I will  try what you have suggested.